File: src/GSLib/chatSystem/GM/chatSystemData.cpp

#include <GSLib/chatSystem/GM/chatSystemData.h>
#include <BSLib/utility/stream.h>

namespace GSLib
{

namespace ChatSystem
{

namespace GM
{

CChatEquipID::CChatEquipID()
:m_bagIndex(-1)
,m_equipID(-1)
{

}

CChatEquipID::~CChatEquipID()
{

}

void CChatEquipID::serializeFrom(BSLib::Utility::CStream& stream) const
{
	stream >> m_bagIndex;
	stream >> m_equipID;
}

}//GM

}//ChatSystem

}//GSLib

File: src/Client/UnityGMClient/ChatSystem/ChatSystemData.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace UnityGMClient
{
    public class CChatEquipID
    {
        public CChatEquipID(UInt32 bagIndex, UInt32 equipID)
        {
            m_bagIndex = bagIndex;
            m_equipID = equipID;
        }
        public UInt32 m_bagIndex;
        public UInt32 m_equipID;

        public void serializeTo(UnityUtility.CStream stream)
        {
            stream.Write(m_bagIndex);
            stream.Write(m_equipID);
        }
    }

}
